elk/extraction/extraction.py

```diff
@@ -163,54 +174,55 @@ def extract_hiddens(
 
             # Iterate over answers
             for j, choice in enumerate(record):
-                text = choice["text"]
-
-                # TODO: Do something smarter than "rindex" here. Really we want to
-                # get the span of the answer directly from Jinja, but that doesn't
-                # seem possible. This approach may fail for complex templates.
-                answer_start = text.rindex(choice["answer"])
+                text = choice["question"]
 
                 # Only feed question, not the answer, to the encoder for enc-dec models
-                if model.config.is_encoder_decoder:
-                    # TODO: Maybe make this more generic for complex templates?
-                    text = text[:answer_start].rstrip()
-                    target = choice["answer"]
-                else:
-                    target = None
+                target = choice["answer"] if is_enc_dec else None
 
                 # Record the EXACT string we fed to the model
                 variant_inputs.append(text)
-                inputs = tokenizer(
+                encoding = tokenizer(
                     text,
-                    return_offsets_mapping=True,
+                    add_special_tokens=False,
                     return_tensors="pt",
                     text_target=target,  # type: ignore[arg-type]
                     truncation=True,
-                )
+                ).to(device)
+                input_ids = assert_type(Tensor, encoding.input_ids)
 
-                # The offset_mapping is a sorted list of (start, end) tuples. We locate
-                # the start of the answer in the tokenized sequence with binary search.
-                offsets = inputs.pop("offset_mapping").squeeze().tolist()
-                inputs = inputs.to(device)
+                if is_enc_dec:
+                    answer = assert_type(Tensor, encoding.labels)
+                else:
+                    encoding2 = tokenizer(
+                        choice["answer"],
+                        add_special_tokens=False,
+                        return_tensors="pt",
+                    ).to(device)
+                    answer = assert_type(Tensor, encoding2.input_ids)
+
+                    input_ids = torch.cat([input_ids, answer], dim=-1)
```

Comment:
I don't believe `question` ends with whitespace nor `answer` begins with whitespace, so isn't the input text slightly malformed?

Reply:
No, because the tokenizer doesn't explicitly represent the whitespace anyway. For example, "hello world" is just represented as a token for "hello" followed by a token for "world" in basically every tokenizer I can think of. FWIW `lm-eval-harness` does make the assumption that this is valid to do